src/read.c

```diff
@@ -566,7 +566,7 @@ static avifResult avifCodecDecodeInputFillFromDecoderItem(avifCodecDecodeInput *
             }
         }
         if (remainingSize > 0) {
-            assert(layerCount == 3);
+            AVIF_CHECKERR(layerCount == 3, AVIF_RESULT_INTERNAL_ERROR);
```

Comment:
I only inspected this change, and this is an example of condition that can be safely checked with `assert()`. The reason is that one can prove this condition is true by analyzing only a few lines of code (specifically, line 544 and lines 551-568).

It seems that `AVIF_CHECKERR(..., AVIF_RESULT_INTERNAL_ERROR)` is more appropriate for conditions that can only be proved true by analyzing multiple functions or complicated code.